### PR TITLE
add Wallet API

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

This PR added newly function in wallet api.
- add option enum in WalletQueryOptions
- add newly function in Token History API
- For testing in circleCI, the newly added test is ignored.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/ground-x/caver-java-ext-kas/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/ground-x/caver-java-ext-kas)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public String getType() {
 
         /**
          * Check if there is an enum mapped to the given type string.
-         * @param type The type string to find enum defined in CONTRACT_TYPE
+         * @param kind The type string to find enum defined in CONTRACT_TYPE
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java:103]:
```suggestion
         * @param kind The kind string to find enum defined in CONTRACT_KIND
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public String getType() {
 
         /**
          * Check if there is an enum mapped to the given type string.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java:102]:
```suggestion
         * Check if there is an enum mapped to the given kind string.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java [hunk 3]
```diff
@@ -152,7 +152,7 @@ public String getKind() {
...
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java:154]:
```suggestion
         * Check if there is an enum mapped to the given status string.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java [hunk 4]
```diff
@@ -152,7 +152,7 @@ public String getKind() {
...
-         * @param type The type string to find enum defined in LABEL_STATUS
+         * @param status The type string to find enum defined in LABEL_STATUS
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/tokenhistory/TokenHistoryQueryOptions.java:155]:
```suggestion
         * @param status The status string to find enum defined in LABEL_STATUS
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java [hunk 5]
```diff
@@ -1078,6 +1092,317 @@ public Call getAccountCountByKRNAsync(String krn, ApiCallback<AccountCountByKRN>
...
+     * @return Call
+     * @throws ApiException
+     */
+    public Call getKeyAsync(String keyId, ApiCallback<Key> callback) throws ApiException {
+        return getKeyApi().getKeyAsync(chainId, keyId, callback);
+    }
+
+    /**
+     * Sign a message using a key create by KAS. <br>
+     * It used default KRN. <br>
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java:1147]:
```suggestion
     * The default KRN will be used. <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java [hunk 6]
```diff
@@ -1078,6 +1092,317 @@ public Call getAccountCountByKRNAsync(String krn, ApiCallback<AccountCountByKRN>
...
+    public KeySignDataResponse signMessage(String keyId, String data, String krn) throws ApiException {
+        KeySignDataRequest request = new KeySignDataRequest();
+        request.setData(data);
+
+        return getKeyApi().keySignData(chainId, keyId, request, krn);
+    }
+
+    /**
+     * Sign a message using a key create by KAS asynchronously. <br>
+     * It used default KRN. <br>
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java:1179]:
```suggestion
     * The default KRN will be used. <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java [hunk 7]
```diff
@@ -1078,6 +1092,317 @@ public Call getAccountCountByKRNAsync(String krn, ApiCallback<AccountCountByKRN>
...
+    public Call signMessageAsync(String keyId, String data, ApiCallback<KeySignDataResponse> callback) throws ApiException {
+        KeySignDataRequest request = new KeySignDataRequest();
+        request.setData(data);
+
+        return signMessageAsync(keyId, data, null, callback);
+    }
+
+    /**
+     * Sign a message using a key create by KAS asynchronously. <br>
+     * It used default KRN. <br>
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java:1196]:
```suggestion
     * The default KRN will be used. <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java [hunk 8]
```diff
@@ -1078,6 +1092,317 @@ public Call getAccountCountByKRNAsync(String krn, ApiCallback<AccountCountByKRN>
...
+        return getKeyApi().keySignDataAsync(chainId, keyId, request, krn, callback);
+    }
+
+    /**
+     * Register accounts which used before. <br>
+     *
+     * POST /v2/registration/account
+     * @param request The AccountRegistrationRequest instance contains account informations to be registered in KAS <br>
+     *                The rlp field of AccountRegistration should be set as an encoded FeeDelegatedAccountUpdate(without set a fee payer and fee payer signature) using the key and the address to be registered in KAS. <br>
+     * @return RegistrationStausResponse
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java:1218]:
```suggestion
     * @return RegistrationStatusResponse
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java [hunk 9]
```diff
@@ -1078,6 +1092,317 @@ public Call getAccountCountByKRNAsync(String krn, ApiCallback<AccountCountByKRN>
...
+    public RegistrationStatusResponse registerAccounts(AccountRegistrationRequest request) throws ApiException {
+        return registerAccounts((List<AccountRegistration>)request);
+    }
+
+    /**
+     * Register accounts which used before. <br>
+     * POST /v2/registration/account
+     * @param request The List of account information to be registered in KAS. <br>
+     *                The rlp field of AccountRegistration should be set as an encoded FeeDelegatedAccountUpdate(without set a fee payer and fee payer signature) using the key and the address to be registered in KAS. <br>
+     * @return RegistrationStausResponse
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java:1230]:
```suggestion
     * @return RegistrationStatusResponse
```

File: src/main/java/xyz/groundx/caver_ext_kas/rest_client/io/swagger/client/api/wallet/api/RegistrationApi.java [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+        
+        
+    }
+
+    /**
+     * Account Registration
+     * Register account which used before
+     * @param xChainId Klaytn chain network ID (1001 or 8217) (required)
+     * @param body  (optional)
+     * @return RegistrationStausResponse
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/rest_client/io/swagger/client/api/wallet/api/RegistrationApi.java:130]:
```suggestion
     * @return RegistrationStatusResponse
```

File: src/main/java/xyz/groundx/caver_ext_kas/rest_client/io/swagger/client/api/wallet/api/RegistrationApi.java [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+        ApiResponse<RegistrationStatusResponse> resp = registerAccountWithHttpInfo(xChainId, body);
+        return resp.getData();
+    }
+
+    /**
+     * Account Registration
+     * Register account which used before
+     * @param xChainId Klaytn chain network ID (1001 or 8217) (required)
+     * @param body  (optional)
+     * @return ApiResponse&lt;RegistrationStausResponse&gt;
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/rest_client/io/swagger/client/api/wallet/api/RegistrationApi.java:143]:
```suggestion
     * @return ApiResponse&lt; RegistrationStatusResponse&gt;
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletQueryOptions.java [hunk 12]
```diff
@@ -16,12 +16,66 @@
...
+
+        String status;
+
+        ACCOUNT_STATUS(String status) {
+            this.status = status;
+        }
+
+        /**
+         * Check if there is an enum mapped to the given type string.
+         * @param status The type string to find enum defined in LABEL_STATUS
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletQueryOptions.java:43]:
```suggestion
         * Check if there is an enum mapped to the given status string.
         * @param status The status string to find enum defined in LABEL_STATUS
```

File: src/test/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletAPITest.java [hunk 13]
```diff
@@ -2739,4 +2745,308 @@ public void checkResponse() throws ExecutionException, InterruptedException {
...
+
+        AccountRegistration registration = new AccountRegistration();
+        registration.setAddress(account.getAddress());
+        registration.setKeyId(response.getItems().get(0).getKeyId());
+        registration.setRlp(createAccountUpdateTx(keyring, response.getItems().get(0).getPublicKey()).getRLPEncoding());
+
+        AccountRegistrationRequest request = new AccountRegistrationRequest();
+        request.add(registration);
+
+        RegistrationStatusResponse result = caver.kas.wallet.registerAccounts(request);
```
Comment (jimni1222) [inline src/test/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletAPITest.java:2988]: How about checking status in result?
If registration is fail, KAS returns failure in response object.

Comment (sirano11) [reply src/test/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletAPITest.java:2988]: Okay, I added logic that check registration status.

**Review by jimni1222 — APPROVED** [review]
LGTM =) !!

**Pull request merged**